Shopify/shopify_app
# Fixing embedded_redirect_url redirect loop

### What this PR does

There are some (quite) edge case scenarios where the `embedded_redirect_url` config will trigger a 302 loop to itself, when reloading the app in the admin after a scope change, in production (or SSR) mode.

To avoid that issue, we will simply not redirect in that scenario, so that the request is considered valid and reaches the backend - there is an assumption here that this page does not contain sensitive information and does not handle actual data, but _just_ blindly redirects.

### Reviewer's guide to testing

1. Create a ruby app with the CLI and follow the steps to run it in dev mode. Open the app to install it and kill the server, keeping the browser window open.
2. Run `npm run build --api-key=<apiKey>`
3. Run `ngrok http 3000` on a window
4. Run `cd web` in the app
5. Run `RAILS_SERVE_STATIC_FILES=1 SHOPIFY_API_KEY=<apiKey> SHOPIFY_API_SECRET=<apiSecret> SCOPES=write_products,read_orders HOST=<ngrokAddress> rails server -b 0.0.0.0 -e production -p 3000`
6. Refresh the admin page now that the scopes have changed.
  1. With the current app, you'll see the iframe redirecting internally to login and borking
  2. With this PR, it should break out of the iframe

### Checklist

Before submitting the PR, please consider if any of the following are needed:
